Add tab completion script for zsh and fix bash completion for ABS services

scotje opened the pull request:
## Status

Ready for Merge

## Description

Adds a new `--hostnameonly` option to `floaty list` to serve as more reliable plumbing for the completion scripts, then updates the bash completion script to use that option and implements a new zsh completion script with equivalent functionality to the bash script.

## Reviewers

@puppetlabs/dio
@highb
@briancain

briancain commented on lib/vmfloaty/utils.rb:
```diff
@@ -78,27 +78,44 @@ def self.generate_os_hash(os_args)
+    when 'ABS'
+      host_data['allocated_resources'].each do |vm_name, _i|
+        puts vm_name['hostname']
+      end
```
It's better here to build up this list of hostnames, and `puts` once.

scotje replied:
Oops, I misunderstood, just one second.

scotje replied:
There we go.

scotje commented:
@puppetlabs/dio if you all have write access would you mind reviewing/merging when you have a chance?

highb reviewed (approved):
The `--hostnameonly` flag is a nice general addition. Perhaps a fish shell completion script is coming soon? 😆

scotje commented:
I rebased this and fixed the tests up as best I could but ran out of time to completely figure out how the default output format of ABS hosts had changed and what I needed to mock out to make those tests pass. If someone has time to fix the tests and wants to re-push this to a new PR, I'm fine with that. Otherwise I can take another look tomorrow.

genebean commented:
@sbeaulie is this maybe something you have some insight into? Also, does anything here need adjusting as a result of #99 that I just merged?

sbeaulie commented:
yes looking into fixing the tests now

sbeaulie commented:
I also found a bug and corrected it so that hybrid requests to ABS (vmpooler, nspooler) can pretty print the hostnames properly.
